sirepo/http_request.py

```diff
@@ -80,6 +80,8 @@ def parse_post(**kwargs):
         model (object): `parse_model`
         name (object): `parse_name`
         template (object): `sirepo.template.import_module`
+        require_sim_type (bool): `require_sim_type`
```

Comment:
This parameter should go away. The problem is in require_sim_type():

```py
def require_sim_type(sim_type):
    if sim_type not in sirepo.feature_config.auth_controlled_sim_types():
        return
    if not _is_logged_in():
        # If a user is not logged in, we allow any sim_type, because
        # the GUI has to be able to get access to certain APIs before
        # logging in.
        return
    check_user_has_role(
        logged_in_user(),
        sirepo.auth_role.for_sim_type(sim_type),
    )
```

Comments are a good indication of a problem. If the user is not logged in, we need to return the minimal schema that allows the user to login. The app will then need to refresh itself. I think that's what happens now when you login anyway. 

I think a minimal schema is too complicated for the moment. (This is what happens when you bolt on security, instead of designing it in.)  

What we should do is add a special decorator that documents the APIs similar to who api_perm works. It could be called something `@allow_unauthorized_sim_type`. 

In uri_router, we save the current api in srcontext. You can see how to do that in http_request, for example. There's already bracketing code.

Inside auth.require_sim_type, if the type is unauthorized, it would allow the API to be authorized.  

As far as check_auth_method, this would happen in require_sim_type, not where it is now. If it is a moderated sim type, then Forbidden is raised in required_sim_type if the user is logged in, and the user doesn't have a user_name that's an email.